src/script/page/RightSidebar/MessageDetails/MessageDetails.tsx

```diff
@@ -119,15 +133,32 @@ const MessageDetails: FC<MessageDetailsProps> = ({
       return receiptUsers.length ? MESSAGE_STATES.RECEIPTS : MESSAGE_STATES.NO_RECEIPTS;
     }
 
-    return likeUsers.length ? MESSAGE_STATES.LIKES : MESSAGE_STATES.NO_LIKES;
-  }, [supportsReceipts, isReceiptsOpen, messageEntity, receiptUsers, likeUsers]);
-
-  const getLikes = useCallback(async (reactions: UserReactionMap) => {
-    const currentLikes = Object.keys(reactions);
-    const usersLikes = await userRepository.getUsersById(currentLikes.map(likeId => ({domain: '', id: likeId})));
-    const sortedUsersLikes = usersLikes.sort(sortUsers);
+    return getTotalReactionUsersCount(reactionUsers) ? MESSAGE_STATES.REACTIONS : MESSAGE_STATES.NO_REACTIONS;
+  }, [supportsReceipts, isReceiptsOpen, messageEntity, receiptUsers, reactionUsers]);
+
+  const getReactions = useCallback(async (reactions: UserReactionMap) => {
+    const usersMap: Record<string, User> = {};
+    const currentReactions = Object.keys(reactions);
+    const usersReactions = await userRepository.getUsersById(
+      currentReactions.map(reactionId => ({domain: '', id: reactionId})),
+    );
+    usersReactions.forEach(user => {
+      usersMap[user.id] = user;
+    });
+    const reactionsGroupByUser: {[key: string]: User[]} = {};
+    for (const userId in reactions) {
+      const user = usersMap[userId];
+      const userReactions = reactions[userId] && reactions[userId].split(',');
+      for (const reaction of userReactions) {
+        if (reactionsGroupByUser[reaction]) {
+          reactionsGroupByUser[reaction].push(user);
+        } else {
+          reactionsGroupByUser[reaction] = [user];
+        }
+      }
+    }
```

Comment:
This seems to be duplicated code. I believe the `transformReactionObj` does exactly this, isn't it?